carbon-design-system/carbon
# refactor(ui-shell): update ui shell tests to use testing library

Closes #11558 

This PR updates our UI Shell components to use testing library

#### Changelog

**New**

**Changed**

- [x] Content
- [x] HeaderContainer
- [x] HeaderGlobalAction
- [x] HeaderGlobalBar
- [x] HeaderMenu
- [x] HeaderMenuButton
- [x] HeaderMenuItem
- [x] HeaderName
- [x] HeaderNavigation
- [x] Link
- [x] SideNav
- [x] SideNavDetails
- [x] SideNavFooter
- [x] SideNavHeader
- [x] SideNavIcon
- [x] SideNavItem
- [x] SideNavItems
- [x] SideNavLink
- [x] SideNavLinkText
- [x] SideNavMenu
- [x] SideNavMenuItem
- [x] SideNavSwitcher
- [x] SkipToContent

#### Testing / Reviewing

- Verify all tests use React testing library
- Verify tests use our test conventions with RTL
